File: FMI/ANUL II/SEM 2/MPP/ProiectJava/Zboruri/ZboruriModel/src/main/java/ro/mpp2024/domain/Bilet.java
package ro.mpp2024.domain;

import java.util.List;

public class Bilet extends Entity<Integer> {
    private String Client;
    private String Oras;
    private String Tara;
    private List<String> Turisti;
    private int NrLocuri;
    private Zbor Zbor;

    public Bilet(String client, String oras, String tara, List<String> turisti, int nrLocuri, Zbor zbor) {
        this.Client = client;
        this.Oras = oras;
        this.Tara = tara;
        this.Turisti = turisti;
        this.NrLocuri = nrLocuri;
        this.Zbor = zbor;
    }

    public String getClient() {
        return Client;
    }

    public void setClient(String client) {
        this.Client = client;
    }

    public String getOras() {
        return Oras;
    }

    public void setOras(String oras) {
        this.Oras = oras;
    }

    public String getTara() {
        return Tara;
    }

    public void setTara(String tara) {
        this.Tara = tara;
    }

    public List<String> getTuristi() {
        return Turisti;
    }

    public void setTuristi(List<String> turisti) {
        this.Turisti = turisti;
    }

    public int getNrLocuri() {
        return NrLocuri;
    }

    public void setNrLocuri(int nrLocuri) {
        this.NrLocuri = nrLocuri;
    }

    public Zbor getZbor() {
        return Zbor;
    }

    public void setZbor(Zbor zbor) {
        this.Zbor = zbor;
    }

    @Override
    public String toString() {
        return "Bilet{" +
                "client='" + Client + '\'' +
                ", turisti=" + Turisti +
                ", nrLocuri=" + NrLocuri +
                ", zbor=" + Zbor +
                '}';
    }
}

File: FMI/ANUL II/SEM 2/MPP/ProiectCSharp/Zboruri/Model/Bilet.cs
namespace Model
{
    [Serializable]
    public class Bilet:Entity<int>
    {
        public string Client { get; set; }
        public string Oras { get; set; }
        public string Tara { get; set; }
        public List<TuristBilet> Turisti { get; set; }
        public int NrLocuri { get; set; }
        public Zbor Zbor { get; set; }
        
        public Bilet()
        {
        }

        public Bilet(string client, string oras, string tara, List<TuristBilet> turisti, int nrLocuri, Zbor zbor)
        {
            Client = client;
            Oras = oras;
            Tara = tara;
            Turisti = turisti;
            NrLocuri = nrLocuri;
            Zbor = zbor;
        }

        public override string ToString()
        {
            return $"{nameof(Client)}: {Client}, {nameof(Oras)}: {Oras}, {nameof(Tara)}: {Tara}, {nameof(Turisti)}: {Turisti}, {nameof(NrLocuri)}: {NrLocuri}, {nameof(Zbor)}: {Zbor}";
        }
    }
}